src/gui_common/TweakedColourPicker.cs

```diff
@@ -0,0 +1,310 @@
+        GetChild(4).GetChild<Control>(4).Hide();
+        GetChild<Control>(5).Hide();
+        GetChild<Control>(6).Hide();
+        GetChild<Control>(7).Hide();
+
+        pickerButton = GetChild(1).GetChild<ToolButton>(1);
+        hsvCheckButton = GetNode<CheckButton>("ButtonsContainer/HSVCheckButton");
+        rawCheckButton = GetNode<CheckButton>("ButtonsContainer/RawCheckButton");
+        hexColorEdit = GetNode<LineEdit>("ButtonsContainer/HexColorEdit");
+        hexColorEdit.Text = "ffffff";
+        separator = GetNode<HSeparator>("Separator");
+        presetsContainer = GetNode<GridContainer>("PresetContainer");
+        addPresetButton = GetNode<Button>("PresetContainer/AddPresetButton");
+
+        // Update control state.
+        HSVButtonEnabled = hsvButtonEnabled;
+        RawButtonEnabled = rawButtonEnabled;
+        HsvMode = base.HsvMode;
+        RawMode = base.RawMode;
+        PresetsEnabled = presetsEnabled;
+        PresetsVisible = presetsVisible;
+
+        // Load initial presets.
+        foreach (var presetsStored in PresetsStorage)
+        {
+            if (presetsStored.Key == GetPath())
+            {
+                foreach (var color in presetsStored.Value)
+                    AddPreset(color);
+
+                PresetsStorage.Remove(presetsStored);
+                break;
+            }
+        }
+
+        Translate();
+    }
+
+    public override void _ExitTree()
+    {
+        // Store presets.
+        PresetsStorage.Add(new KeyValuePair<string, Color[]>(GetPath(), GetPresets()));
+
+        base._ExitTree();
+    }
+
+    public override void _Notification(int what)
+    {
+        if (what == NotificationTranslationChanged)
+            Translate();
+
+        base._Notification(what);
+    }
+
+    public new void AddPreset(Color color)
+    {
+        foreach (var knownPreset in presets)
+        {
+            if (knownPreset.Color == color)
+                return;
+        }
+
+        AddPresetLocal(color);
+
+        base.AddPreset(color);
+    }
+
+    public new void ErasePreset(Color color)
+    {
+        foreach (var preset in presets)
+        {
+            if (preset.Color == color)
+            {
+                OnPresetDeleted(preset);
+                break;
+            }
+        }
+    }
+
+    private void Translate()
+    {
+        pickerButton.HintTooltip = TranslationServer.Translate("COLOR_PICKER_PICK_COLOR");
+        addPresetButton.HintTooltip = TranslationServer.Translate("COLOR_PICKER_ADD_PRESET");
+        TweakedColorPickerPreset.HintTooltipBase = TranslationServer.Translate("COLOR") + ": #{0}\n"
+            + TranslationServer.Translate("LEFT_MOUSE") + ": "
+            + TranslationServer.Translate("COLOR_PICKER_SELECT_PRESET") + "\n"
+            + TranslationServer.Translate("RIGHT_MOUSE") + ": "
+            + TranslationServer.Translate("COLOR_PICKER_DELETE_PRESET");
+    }
+
+    private void AddPresetLocal(Color color)
```

Comment:
Doesn't sound quite obvious to me what `AddPresetLocal` means.

Reply:
Once I thought I may call local add preset logic instead of call a whole logic (without affecting ColorPicker), but later I removed that code.

This has been removed.